### PR TITLE
Added support  persistence_config field to google_redis_instance resource.

### PR DESCRIPTION
**Pull request by googlerjk**
Added support  persistence_config field to 'google_redis_instance' resource.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note: enhancement
redis: Added `persistence_config` field to the `google_redis_instance` resource.

```

### CODE REVIEW

**Comment by c2thorn** [comment]
`rdb_snapshot_start_time` needs to default to api as well?

File: mmv1/products/redis/api.yaml [hunk 1]
```diff
@@ -143,6 +143,57 @@ objects:
...
+        name: persistenceConfig
+        description: Maintenance policy for an instance.
+        properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'persistenceMode'
+            required: true
+            description: |
+              Optional. Controls whether Persistence features are enabled. If not provided, the existing value will be used.
+
+              - PERSISTENCE_MODE_UNSPECIFIED:	Not set.
```
Comment (c2thorn) [inline mmv1/products/redis/api.yaml:156]: Can we remove the `PERSISTENCE_MODE_UNSPECIFIED` and `SNAPSHOT_PERIOD_UNSPECIFIED` values/references?

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/products/redis/api.yaml (hunk 1 above)
Comment (googlerjk) [reply mmv1/products/redis/api.yaml]: done

**Review by c2thorn — APPROVED** [review]

**Comment by jcanseco** [comment]
Hi @googlerjk @c2thorn. Quick question: Is the `rdb_snapshot_period` field supposed to be optional, not required? It's optional in the [API docs](https://cloud.google.com/memorystore/docs/redis/reference/rest/v1/projects.locations.instances#persistenceconfig). Just want to check to make sure I understand correctly.

**Comment by c2thorn** [comment]
> Hi @googlerjk @c2thorn. Quick question: Is the `rdb_snapshot_period` field supposed to be optional, not required? It's optional in the [API docs](https://cloud.google.com/memorystore/docs/redis/reference/rest/v1/projects.locations.instances#persistenceconfig). Just want to check to make sure I understand correctly.

Looks like a miss on my review. Our own documentation states it as optional. Luckily `required` -> `optional` is a non-breaking change. @googlerjk Can you correct this in a new PR?

**Comment by c2thorn** [comment]
@jcanseco that would be helpful, thank you!

**Comment by c2thorn** [comment]
@jcanseco that would be helpful, thank you!

**Comment by jcanseco** [comment]
Done: https://github.com/hashicorp/terraform-provider-google/issues/12854

**Pull request merged**